src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimal(animal) {
+    Animal.alive.splice(Animal.alive.indexOf(animal), 1);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden !== true) {
```

Comment:
probably you mean this? 😄 
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
```